ibis/backends/pandas/execution/window.py

```diff
@@ -32,19 +33,27 @@
 
 
 def _post_process_empty(
-    result: Any, parent: pd.DataFrame, order_by: List[str], group_by: List[str]
+    result: Any,
+    parent: pd.DataFrame,
+    order_by: List[str],
+    group_by: List[str],
+    timecontext: Optional[TimeContext],
 ) -> pd.Series:
     assert not order_by and not group_by
     if isinstance(result, pd.Series):
         # `result` is a Series when an analytic operation is being
         # applied over the window, since analytic operations are N->N
+        if timecontext:
```

Comment:
Why do we need this here if this is already handled inside aggcontext?

Reply:
We need this since it's not fully handled in aggcontext.

In `aggcontext.py`, for class `Window`, method `agg`, there are 3 different execution paths:
1. built-in aggregation, this is handled by `window_agg_built_in` in line 649, it returns a multiindex series and context adjustment(trimming) works well.
2. udfs, this is handled by `window_agg_udf` in line 634, and it is the main change we made. 
3. However there is another shortcut, in line 547, for cases where there is not groupby nor orderby. It is just calling pandas methods, the result could be either a series(analytic operations), or just a number(reduction operations). This `post_process_empty` here is meant to deal with this case, to return the series, or extend the result into a series. In order for this series to be trimmed, we need to add time index here.

Reply:
I see. Can you add comments here to explain?

Reply:
Sure, I added an inline comment for this